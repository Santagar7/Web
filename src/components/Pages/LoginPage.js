//import '../../css/LoginPage.css';
import {Link, useNavigate} from "react-router-dom";
import React from 'react';
import axios from "axios";
axios.defaults.withCredentials = true;

const url = 'http://localhost:7000/';

export default function LoginPage() {
    const navigator = useNavigate();

    function Login(event){
        event.preventDefault();
        let name = document.getElementById('name').value;
        let password = document.getElementById('password').value;
        let elem = document.getElementById('login');
        if(password===''||name===''){
            elem.textContent = 'Invalid data';
            elem.style.color = 'crimson';
            return;
        }

        axios.post(url,{
            'name':name,
            'password':password
        }).then((resp)=>{
            if(resp.data.message==='Success'){
                navigator("/budgets");
            }
            else{
                elem.textContent = resp.data.message;
                elem.style.color = 'crimson';
            }
        })
    }

    return (
        <div className="center">
            <h1 id="login">Login</h1>
            <form className="main">
                <div className="data">
                    <input id='name'type="text" data-testid='name' placeholder=" " autoComplete="on"/>
                    <span></span>
                    <label>UserName</label>
                </div>
                <div className="data">
                    <input id="password" type="password" data-testid='password' placeholder=" " autoComplete="on"/>
                    <span></span>
                    <label>Password</label>
                </div>

                <input type="submit" value ="Login" onClick={Login} data-testid='submit'/>
                <div className="signup"> First time here ? 
                    <Link to='/signup'>Register</Link>
                </div>
            </form>
        </div>
    );
}
