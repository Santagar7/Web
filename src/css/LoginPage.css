body {
    display: flex;
    margin: 0;
    padding: 0;
    background-image: radial-gradient(circle, rgba(238,174,202,1) 0%, rgba(241,198,186,1) 34%, rgba(148,187,233,1) 100%);
    height: 100vh;
    font-family: arial;
    overflow: hidden;
  }

.center {
    margin: 180px auto !important;
    width: 440px;
    position: relative;
    background: white;
    border-radius: 10px;
    padding-bottom: 35px;
    z-index: 5;
}
.center h1{
    text-align: center;
    padding: 5px 0 10px 0;
    border-bottom: 1px solid silver;
    user-select: none;
    border-radius: 10px 10px 0 0;
}
.center .main {
    padding: 0 60px;
    box-sizing: border-box;
}

.main .data {
    position: relative;
    border-bottom: 2px solid silver;
    margin: 30px 0;
}

.data input {
    width: 100%;
    padding: 0 5px;
    height: 20px;
    border: none;
    font-size: 14px;
    background: none;
    outline: none;
}
.data label {
    position: absolute;
    top: 50%;
    left: 5px;
    color: silver;
    transition: 0.2s;
    transform: translateY(-50%);
    font-size: 14px;
    pointer-events: none;
    user-select: none;
}
.data span::before {
    content: "";
    position: absolute;
    top: 20px;
    left: 0;
    width: 0%;
    height: 2px;
    background: blue;
    transition: 0.5s;
}
.data input:focus ~ label, .data input:not(:placeholder-shown) ~ label {
    top: -5px;
    color: blue;
}
.data input:focus ~ span::before, .data input:not(:placeholder-shown) ~ span::before {
    width: 100%;
}
input[type=submit] {
    width: 100%;
    height: 30px;
    border: 1px solid silver;
    background: HoneyDew;
    border-radius: 25px;
    font-size: 16px;
    color: blue;
    font-weight: 700;
    cursor: pointer;
    outline: none;
}
input[type=submit]:hover {
    border-color: blue;
    transition: 0.4s;
}

.signup {
    margin: 30px 0;
    font-size: 18px;
    color: black;
  }
  .signup a {
    color: blue;
    text-decoration: none;
    cursor: pointer;
  }
  .signup a:hover {
    color: blue;
    text-decoration: underline;
  }