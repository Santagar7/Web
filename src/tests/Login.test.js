import { BrowserRouter } from 'react-router-dom';
import { expect, test } from '@jest/globals';
import { render, screen, fireEvent } from '@testing-library/react';
import LoginPage from '../components/Pages/LoginPage';
import renderer from 'react-test-renderer';

test('should render correctly', () => {
    const tree = renderer.create(<BrowserRouter><LoginPage/></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
test('test login data', () => {
    render(
        <BrowserRouter>
            <LoginPage />
        </BrowserRouter>,
    );
    const inputName = screen.getByTestId('name');
    fireEvent.change(inputName, { target: { value: 'steve' } });
    expect(inputName.value).toBe('steve');

    const inputPass = screen.getByTestId('password');
    fireEvent.change(inputPass, { target: { value: 'steve' } });
    expect(inputName.value).toBe('steve');
});

test('test submit btn', () => {
    render(
        <BrowserRouter>
            <LoginPage />
        </BrowserRouter>,
    );
    const submit = screen.getByTestId('submit');
    fireEvent.click(submit);
});