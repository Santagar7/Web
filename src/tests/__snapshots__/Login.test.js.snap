// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="center"
>
  <h1
    id="login"
  >
    Login
  </h1>
  <form
    className="main"
  >
    <div
      className="data"
    >
      <input
        autoComplete="on"
        data-testid="name"
        id="name"
        placeholder=" "
        type="text"
      />
      <span />
      <label>
        UserName
      </label>
    </div>
    <div
      className="data"
    >
      <input
        autoComplete="on"
        data-testid="password"
        id="password"
        placeholder=" "
        type="password"
      />
      <span />
      <label>
        Password
      </label>
    </div>
    <input
      data-testid="submit"
      onClick={[Function]}
      type="submit"
      value="Login"
    />
    <div
      className="signup"
    >
       First time here ?
      <a
        href="/signup"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </form>
</div>
`;
